packages/vite/src/node/config.ts

```diff
@@ -754,15 +769,17 @@ export async function resolveConfig(
     isWorker: true,
     mainConfig: resolved,
   }
-  resolvedConfig.worker.plugins = await resolvePlugins(
-    workerResolved,
-    workerPrePlugins,
-    workerNormalPlugins,
-    workerPostPlugins,
-  )
+
+  const getResolvedWorkerPlugins = async () => {
+    return resolvePlugins(
+      workerResolved,
+      ...(await getSortedWorkerUserPlugins(rawWorkerUserPlugins)),
+    )
+  }
+
   Object.assign(
     resolvedConfig.worker,
-    createPluginHookUtils(resolvedConfig.worker.plugins),
+    createPluginHookUtils(await getResolvedWorkerPlugins()),
```

Comment:
Might need to add some comments. The assumption is that the user `worker.plugins` function would be side effect free. This allows us to pre-process relevant hook/config updates but still allow fresh plugin instantiation per worker build. 

Does that make sense? 😅